engine/index.js

```diff
@@ -298,6 +299,107 @@ function populateCanIUsePercent(canIUseData, features) {
   });
 }
 
+const statusFields = ['firefox_status', 'spec_status', 'opera_status',
+                      'webkit_status', 'ie_status'];
+// checking for changes in 'status' object
+function checkForNewData(features, dbTestNumber) {
+  return redis.getClient(dbTestNumber)
+  .then((client) => new Promise((resolve, reject) => {
+    client.get('status', (err, oldStatus) => {
+      try {
+        oldStatus = JSON.parse(oldStatus);
+      } catch (e) {
+        console.log(e);
+      }
+      if (err) {
+        console.error('ERROR: ' + err);
+        reject(err, client);
+      }
+      if (!oldStatus) {
+        oldStatus = {};
+      }
+      features.map((feature) => {
+        feature.updated = {};
+        if (!oldStatus[feature.slug]) {
+          feature.just_started = true;
+        } else {
+          // XXX: check if that can happen outside of test...
+          if (feature.just_started) {
+            delete feature.just_started;
+          }
+          statusFields.forEach((name) => {
+            if (feature[name] !== oldStatus[feature.slug][name]) {
+              feature.updated[name] = {
+                from: oldStatus[feature.slug][name],
+                to: feature[name],
+              };
+            }
+          });
+        }
+      });
+      resolve(client);
+    });
+  })).then((client) => {
+    client.quit();
+    return features;
+  }).catch((err, client) => {
+    console.error('ERROR:', err);
+    client.quit();
+  });
+}
+
+// `status` key holds an Object representation of `status.json`
+// `changed` is a hashtag with just changed data stored by date
+function saveData(features, dbTestNumber) {
+  // store changes under date
+  const date = new Date().toISOString();
+  const statusData = {};
+  const changedData = {};
+  features.map((feature) => {
+    statusData[feature.slug] = feature;
+    if (Object.keys(feature.updated).length > 0) {
+      if (!changedData.updated) {
+        changedData.updated = {};
+      }
+      changedData.updated[feature.slug] = feature.updated;
+    }
+    if (feature.just_started) {
+      if (!changedData.started) {
+        changedData.started = [];
+      }
+      changedData.started.push(feature);
+    }
+  });
+  return redis.getClient(dbTestNumber)
+  .then((client) => new Promise((resolve, reject) => {
+    client.set('status', JSON.stringify(statusData), (errS) => {
+      if (errS) {
+        return reject(errS, client);
+      }
+
+      if (!(changedData.updated || changedData.started)) {
+        console.log('DEBUG: no changes found');
+        return resolve(client);
+      }
+      console.log('DEBUG: found new changes');
+      client.hmset('changelog', date, JSON.stringify(changedData), (errC) => {
+        if (errC) {
+          reject(errC, client);
+        }
+        resolve(client);
+      });
+    });
+  }))
+  .catch((err, client) => {
+    console.error('ERROR:', err);
+    client.quit();
+  })
+  .then((client) => {
```

Comment:
If data isn't saved we want to terminate... This should never happen. catch is only used to close the connection to redis - otherwise it would just hang there.